crates/uv-python/src/discovery.rs

```diff
@@ -335,7 +339,24 @@ fn python_executables_from_installed<'a>(
                         }
                     })
                     .inspect(|installation| debug!("Found managed installation `{installation}`"))
-                    .map(|installation| (PythonSource::Managed, installation.executable(false))))
+                    .map(move |installation| {
+                        (
+                            PythonSource::Managed,
+                            if version.patch().is_some() || is_alternative_implementation {
+                                installation.executable(false)
+                            } else {
```

Comment:
When deriving an executable from an installation, if the version is not a patch and it's a CPython installation, uv looks for a minor version symlink directory. If one is found, it will use the symlink-directory containing path. This enables transparently upgradeable virtual environments created with the `venv` module.